test_unstructured/partition/html/test_parser.py

```diff
@@ -0,0 +1,600 @@
+        }
+
+    def and_its_children_are_also_annotated_with_bold_emphasis(self):
+        b = etree.fromstring("<b>rhombus <i>pentagon</i></b>", html_parser).xpath(".//b")[0]
+
+        text_segments = b.iter_text_segments()
+
+        text, annotation = next(text_segments)
+        assert text == "rhombus "
+        assert annotation == {
+            "emphasized_text_contents": "rhombus",
+            "emphasized_text_tags": "b",
+        }
+        text, annotation = next(text_segments)
+        assert text == "pentagon"
+        assert annotation == {
+            "emphasized_text_contents": "pentagon",
+            "emphasized_text_tags": "bi",
+        }
+
+    def but_not_its_tail(self):
+        b = etree.fromstring("<b>rhombus</b> pentagon", html_parser).xpath(".//b")[0]
+
+        text_segments = b.iter_text_segments()
+
+        text, annotation = next(text_segments)
+        assert text == "rhombus"
+        assert annotation == {
+            "emphasized_text_contents": "rhombus",
+            "emphasized_text_tags": "b",
+        }
+        text, annotation = next(text_segments)
+        assert text == " pentagon"
+        assert annotation == {}
+
+
+class DescribeItalic:
+    """Isolated unit-test suite for `unstructured.partition.html.parser.Italic`.
+
+    The `Italic` class is used for `<i>` and `<em>` tags and adds emphasis metadata.
+    """
+
+    def it_annotates_its_text_segment_with_italic_emphasis(self):
+        i = etree.fromstring("<i>rhombus</i>", html_parser).xpath(".//i")[0]
+
+        text_segments = i.iter_text_segments()
+
+        text, annotation = next(text_segments)
+        assert text == "rhombus"
+        assert annotation == {
+            "emphasized_text_contents": "rhombus",
+            "emphasized_text_tags": "i",
+        }
+
+    def and_its_children_are_also_annotated_with_italic_emphasis(self):
+        em = etree.fromstring("<em>rhombus <b>pentagon</b></em>", html_parser).xpath(".//em")[0]
+
+        text_segments = em.iter_text_segments()
+
+        text, annotation = next(text_segments)
+        assert text == "rhombus "
+        assert annotation == {
+            "emphasized_text_contents": "rhombus",
+            "emphasized_text_tags": "i",
+        }
+        text, annotation = next(text_segments)
+        assert text == "pentagon"
+        assert annotation == {
+            "emphasized_text_contents": "pentagon",
+            "emphasized_text_tags": "bi",
+        }
+
+    def but_not_its_tail(self):
+        i = etree.fromstring("<i>rhombus</i> pentagon", html_parser).xpath(".//i")[0]
+
+        text_segments = i.iter_text_segments()
+
+        text, annotation = next(text_segments)
+        assert text == "rhombus"
+        assert annotation == {
+            "emphasized_text_contents": "rhombus",
+            "emphasized_text_tags": "i",
+        }
+        text, annotation = next(text_segments)
+        assert text == " pentagon"
+        assert annotation == {}
```

Comment:
Do we need tests for the LineBreak, RemovedPhrasing, and DefaultElement classes?

Reply:
Yep, good call, added.